test/dynamo/test_base_hop.py

```diff
@@ -73,6 +74,88 @@ def forward(self, l_x_: "f32[3, 3]", l_y_: "f32[3, 3]"):
 """,  # NOQA: B950
         )
 
+    def _find_hop_schema(
+        self, gm: torch.fx.GraphModule, target: Any
+    ) -> list[torch._C.FunctionSchema]:
+        import torch.utils._pytree as pytree
+
+        schemas = []
+        for node in gm.graph.find_nodes(op="call_function", target=target):
+
+            def _get_example_value(node: torch.fx.Node) -> Any:
+                if node.op == "get_attr":
+                    return getattr(gm, node.target)
+                else:
+                    return node.meta["example_value"]
+
+            fake_args, fake_kwargs = pytree.tree_map_only(
+                torch.fx.Node,
+                _get_example_value,
+                (node.args, node.kwargs),
+            )
+            schema = node.target.gen_schema(*fake_args, **fake_kwargs)
+            schemas.append(schema)
+        return schemas
+
+    @torch._dynamo.config.patch(assume_static_by_default=True)
+    def test_schema_gen_single_return(self):
+        def inner(x, y):
+            x.add_(1)
+            y.mul_(-1)
+            return (x @ y).sin().cos()
+
+        x = torch.randn(3, 3, requires_grad=False)
+        y = torch.randn(3, 3, requires_grad=False)
+
+        backend = EagerAndRecordGraphs()
+
+        @torch.compile(backend=backend)
+        def f(x, y):
+            return invoke_quant_test(inner, x, y, scheme="nf4")
+
+        out = f(x.clone(), y)
+        self.assertEqual(out, inner(x.clone(), y))
+        schemas = self._find_hop_schema(backend.graphs[0], invoke_quant_test)
+        self.assertEqual(len(schemas), 1)
+        self.assertExpectedInline(
+            str(schemas[0]),
+            # See Note [schema for signle return item with parenthesis] for why
+            # there's an extra parenthesis in schema of return
+            """invoke_quant_test(Any subgraph, Tensor(a1!) arg0, Tensor(a2!) arg1, str scheme="nf4") -> ((Tensor) out)""",  # noqa: B950
```

Comment:
In aten operators, the output is not actually named. Can we avoid naming the output?

When the output of aten operators is named we actually return a named tuple, which is not the case for HOPs

Reply:
I think this will solve your multiple set of parenthesis problem too. I expect the schema to look like:
```
invoke_quant_test(Any subgraph, Tensor(a1!) arg0, Tensor(a2!) arg1, str scheme="nf4") -> Tensor
```

Reply:
For the return of hop schema, we  have two options: 

1. we could return **"->  Tensor", and "-> Tensor, Tensor"** when return is a tuple of 1 and 2 by constructing the FunctionArugment with returns = [*tuple_output] but we have no way to differentiate tuple of size 1 and a single tensor: they're both **"-> Tensor"** (although for hops we should always return a tuple? there's no confusion if enforce this invariant). 

2. Or we set the returns to be [tuple_output], we'll get **"((Tensor))"** or **"((Tensor, Tensor))"** . In this case, there's no confusion but we have the double bracket.